### PR TITLE
[icu] Fix tool path

### LINKED ISSUE #18086: [icu:x64-windows-static] build failure
**Host Environment**
- OS: Windows 10 21H1
- Compiler: VS 16.9.6

**To Reproduce**
`./vcpkg install icu:x64-windows-static`

**Failure logs** 

```
Computing installation plan...
The following packages will be built and installed:
    icu[core]:x64-windows-static -> 69.1#9
Detecting compiler hash for triplet x64-windows-static...
Could not locate cached archive: C:\Users\Johnathan\AppData\Local\vcpkg\archives\d2\d2df960a9062beb50351d4f0c146928050f82dfe.zip
Starting package 1/1: icu:x64-windows-static
Building package icu[core]:x64-windows-static...
-- Using cached C:/dev/vcpkg/downloads/icu4c-69_1-src.tgz
-- Cleaning sources at C:/dev/vcpkg/buildtrees/icu/src/c-69_1-src-78c1b28057.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/dev/vcpkg/downloads/icu4c-69_1-src.tgz
-- Applying patch C:/dev/vcpkg/ports/icu/disable-escapestr-tool.patch
-- Applying patch C:/dev/vcpkg/ports/icu/remove-MD-from-configure.patch
-- Applying patch C:/dev/vcpkg/ports/icu/fix_parallel_build_on_windows.patch
-- Applying patch C:/dev/vcpkg/ports/icu/fix-extra.patch
-- Using source at C:/dev/vcpkg/buildtrees/icu/src/c-69_1-src-78c1b28057.clean
-- Found external ninja('1.10.2').
-- Configuring x64-windows-static
-- Using msys root at C:/dev/vcpkg/downloads/tools/msys2/d589ebc2087b2b2c
-- Using msys root at C:/dev/vcpkg/downloads/tools/msys2/50864d9fae363d6e
-- Configuring x64-windows-static-dbg
-- Configuring x64-windows-static-rel
-- Using msys root at C:/dev/vcpkg/downloads/tools/msys2/969c0913b9df89e1
-- Building x64-windows-static-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:146 (message):
    Command failed: C:/dev/vcpkg/downloads/tools/msys2/d589ebc2087b2b2c/usr/bin/make.exe -j 9 --trace -f Makefile all
    Working Directory: C:/dev/vcpkg/buildtrees/icu/x64-windows-static-dbg
    See logs for more information:
      C:\dev\vcpkg\buildtrees\icu\build-x64-windows-static-dbg-out.log
      C:\dev\vcpkg\buildtrees\icu\build-x64-windows-static-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_make.cmake:194 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_make.cmake:26 (vcpkg_build_make)
  ports/icu/portfile.cmake:136 (vcpkg_install_make)
  scripts/ports.cmake:141 (include)


Error: Building package icu:x64-windows-static failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: icu:x64-windows-static
  Vcpkg version: 2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9
```

### PR DESCRIPTION
**Pull request by PhoebeHui**
Fixes https://github.com/microsoft/vcpkg/issues/18086

The vcpkg tool path passed when cross build, however, the path doesn't convert to unix path. So when the top level contains 'dev' word, it will cause below failures.

Before:
E:/dev/vcpkg/installed/x64-windows/tools/icu

After:
/e/dev/vcpkg/installed/x64-windows/tools/icu

Failures:
```
cl : Command line warning D9002 : ignoring unknown option '-Xcompiler'
'bash' is not recognized as an internal or external command,
operable program or batch file.
-- return status = 1
Failed to compile oma.c
'LIB.exe' is not recognized as an internal or external command,
operable program or batch file.
-- return status = 1
Error creating Windows DLL library. Failed command: LIB.exe /nologo /out:"../lib/sicudtd.lib" "./out/tmp/icudt69l_dat.obj"
Error generating package data.
make[1]: *** [Makefile:176: packagedata] Error 1
make: *** [Makefile:153: all-recursive] Error 2
```

### CODE REVIEW

File: ports/icu/portfile.cmake [hunk 1]
```diff
@@ -35,7 +35,10 @@ set(DEBUG_TRIPLET ${TARGET_TRIPLET}-dbg)
 
 if(NOT "${TARGET_TRIPLET}" STREQUAL "${HOST_TRIPLET}")
     # cross compiling
-    list(APPEND CONFIGURE_OPTIONS "--with-cross-build=${_VCPKG_INSTALLED_DIR}/${HOST_TRIPLET}/tools/${PORT}")
+    set(TOOL_PATH "${_VCPKG_INSTALLED_DIR}/${HOST_TRIPLET}/tools/${PORT}")
+    # convert to unix path
+    string(REGEX REPLACE "([a-zA-Z]):/" "/\\1/" _VCPKG_TOOL_PATH "${TOOL_PATH}")
```
Comment (JackBoosY) [inline ports/icu/portfile.cmake:40]:
```suggestion
    string(REGEX REPLACE "^([a-zA-Z])+:/" "/\\1/" _VCPKG_TOOL_PATH "${TOOL_PATH}")
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
@Neumann-A @past-due @autoantwort @dg0yt Should we check all the paths in `vcpkg_configure_make` when using mingW crossbuild?

File: ports/icu/portfile.cmake (hunk 1 above)
Comment (JackBoosY) [reply ports/icu/portfile.cmake:40]: I think the best way is to get the drive letter through the cmake function and replace it with unix format (if cmake does not provide a conversion path function).

Comment (JackBoosY) [reply ports/icu/portfile.cmake:40]:
```cmake

            find_program(cygpath NAMES cygpath PATHS "${MSYS_ROOT}/usr/bin" REQUIRED)
            MESSAGE("cygpath: ${cygpath}")
            vcpkg_execute_required_process(
                COMMAND ${cygpath} ${IN_PATHS}
                WORKING_DIRECTORY ${CURRENT_BUILDTREES_DIR}
                OUTPUT_VARIABLE OUT_VARS
                LOGNAME cypath_options
            )
```

Comment (dg0yt) [reply ports/icu/portfile.cmake:40]: Note that accessing `MSYS_ROOT` in the scope of the portfile requires an explicit `vcpkg_acquire_msys`. Wouldn't `find_program` work without `PATHS` option?

Comment (PhoebeHui) [reply ports/icu/portfile.cmake:40]: I'd like to use the regex instead personally. @ras0219 @ras0219-msft, any thoughts?

**Comment by JackBoosY** [comment]
@dg0yt See https://github.com/microsoft/vcpkg/pull/19734.

**Comment by autoantwort** [comment]
> @Neumann-A @past-due @autoantwort @dg0yt Should we check all the paths in `vcpkg_configure_make` when using mingW crossbuild?

I don't think that this is needed. On linux the /dev folder lists all your devices. You never use this folder to put custom stuff there (I don't know it this is even possible)

File: ports/icu/portfile.cmake [hunk 2]
```diff
@@ -35,7 +35,10 @@ set(DEBUG_TRIPLET ${TARGET_TRIPLET}-dbg)
 
 if(NOT "${TARGET_TRIPLET}" STREQUAL "${HOST_TRIPLET}")
     # cross compiling
-    list(APPEND CONFIGURE_OPTIONS "--with-cross-build=${_VCPKG_INSTALLED_DIR}/${HOST_TRIPLET}/tools/${PORT}")
+    set(TOOL_PATH "${_VCPKG_INSTALLED_DIR}/${HOST_TRIPLET}/tools/${PORT}")
```
Comment (ras0219-msft) [inline ports/icu/portfile.cmake:38]:
```suggestion
    set(TOOL_PATH "${CURRENT_HOST_INSTALLED_DIR}/tools/${PORT}")
```

File: ports/icu/portfile.cmake (hunk 1 above)
Comment (PhoebeHui) [reply ports/icu/portfile.cmake]: Updated, thanks for your review!

File: ports/icu/portfile.cmake (hunk 2 above)
Comment (PhoebeHui) [reply ports/icu/portfile.cmake]: Updated!, thanks for your review!

**Review by JackBoosY — APPROVED** [review]

**Review by ras0219-msft — APPROVED** [review]

**Comment by ras0219-msft** [comment]
LGTM once merged from master

**Comment by BillyONeal** [comment]
Thanks for the fix!

**Pull request merged**